tasks/ModelTask.hpp

```diff
@@ -32,24 +33,28 @@ namespace rock_gazebo {
 			
 			void setLinkPorts();
 			void updateLinks();
-			
-			RTT::InputPort<double>* joint_port;
-			typedef std::vector<std::pair<RTT::InputPort<double>*,JointPtr> > JointPort_V;
-			JointPort_V joint_port_list;
-			
-			RTT::InputPort<base::Vector3d>* link_port; 
-			typedef std::vector<std::pair<RTT::InputPort<base::Vector3d>*,LinkPtr> > LinkPort_V;
-			LinkPort_V link_port_list; 
-	        
-	        Joint_V joints;
-	        Link_V links;
-	        
+
+            Joint_V gazebo_joints;
+            Link_V gazebo_links;
+
+            base::samples::Joints joints_in, joints_out;
+
+            typedef std::vector<std::string> NameVector;
+            NameVector link_names;
```

Comment:
You don't need to store this in the class. It is used only once. Make it a local variable.